lms/static/scripts/frontend_apps/components/dashboard/test/AllCoursesActivity-test.js

```diff
@@ -154,50 +164,74 @@ describe('AllCoursesActivity', () => {
     });
   });
 
-  [12, 35, 1, 500].forEach(id => {
-    it('builds expected href for row confirmation', () => {
-      const wrapper = createComponent();
-      const href = wrapper
+  context('when building row confirmation links', () => {
+    function getLinkForID(wrapper, id) {
+      return wrapper
         .find('OrderableActivityTable')
         .props()
         .navigateOnConfirmRow({ id });
+    }
 
-      assert.equal(href, `/courses/${id}`);
+    [12, 35, 1, 500].forEach(id => {
+      it('builds expected links for row confirmation', () => {
+        const wrapper = createComponent();
+        assert.equal(getLinkForID(wrapper, id), `/courses/${id}`);
+      });
+    });
+
+    [
+      {
+        prop: 'students',
+        arg: ['123', '456'],
+        expectedLink: '/courses/123?student_id=123&student_id=456',
+      },
+      {
+        prop: 'assignments',
+        arg: ['123'],
+        expectedLink: '/courses/123?assignment_id=123',
+      },
+      {
+        prop: 'courses',
+        arg: ['11', '33', '88'],
+        expectedLink: '/courses/123', // Selected courses are not propagated
+      },
+    ].forEach(({ prop, arg, expectedLink }) => {
+      it('takes filters into consideration', () => {
```

Comment:
"into consideration" is a bit vague. Perhaps "preserves filters" would be clearer.